### ISSUE TITLE
Crash on commit

### ISSUE THREAD

**Issue by kyrsjo** [spyder-ide/spyder#7979]
### What steps will reproduce the problem?

This was my first commit in a new repo; I had previously ran "git init" in the terminal. I have not (manually) added any files in any way, I assumed that when I clicked "commit" it would ask me which files to commit (which would all of my source files).

### Traceback
```python-traceback
  File "/usr/lib/python3.6/site-packages/spyder/widgets/explorer.py", line 393, in <lambda>
    self.vcs_command(fnames, 'commit')
  File "/usr/lib/python3.6/site-packages/spyder/widgets/explorer.py", line 825, in vcs_command
    for path in sorted(fnames):
TypeError: 'bool' object is not iterable
```

## Versions

* Spyder version: 3.3.1 
* Python version: 3.6.6
* Qt version: 5.10.1
* PyQt5 version: 5.10
* Operating System: Linux 4.17.9-200.fc28.x86_64

### Dependencies

```
pyflakes >=0.6.0  :  1.6.0 (OK)
pycodestyle >=2.3 :  2.3.1 (OK)
pygments >=2.0    :  2.2.0 (OK)
sphinx >=0.6.6    :  1.7.5 (OK)
rope >=0.9.4      :  0.11.0 (OK)
jedi >=0.9.0      :  0.12.0 (OK)
psutil >=0.3      :  5.4.3 (OK)
nbconvert >=4.0   :  5.3.1 (OK)
pandas >=0.13.1   :  0.22.0 (OK)
numpy >=1.7       :  1.14.5 (OK)
sympy >=0.7.3     :  None (NOK)
cython >=0.21     :  0.28.4 (OK)
qtconsole >=4.2.0 :  4.3.1 (OK)
IPython >=4.0     :  6.4.0 (OK)
matplotlib >=2.0.0:  2.2.2 (OK)
pylint >=0.25     :  1.7.5 (OK)
```

**Comment by ccordoba12** [comment]
Thanks for reporting. This is already fixed and it'll be part of our next release (3.3.2).

**Comment by edchepen** [comment]
Has this been fixed on Beta 4.0.0b1  (https://pypi.org/project/spyder/4.0.0b1/)? I just downloaded it and I see the same error. Choosing either "Commit" or "Browse Repository" generates a dialog that says that Spyder has encountered an internal problem. 

 ` File "/home/edm/.virtualenvs/tf3gpu/lib/python3.5/site-packages/spyder/widgets/explorer.py", line 413, in <lambda>
    self.vcs_command(fnames, 'browse')
  File "/home/edm/.virtualenvs/tf3gpu/lib/python3.5/site-packages/spyder/widgets/explorer.py", line 851, in vcs_command
    for path in sorted(fnames):
TypeError: 'bool' object is not iterable`

**Comment by ccordoba12** [comment]
No, it was not. It'll be fixed in beta2.
